packages/discovery-provider/src/api/v1/models/purchasable_content.py

```diff
@@ -0,0 +1,134 @@
+from flask_restx import fields
+from flask_restx.fields import MarshallingError, marshal
+
+from .common import ns
+from .playlists import (
+    full_playlist_model,
+    full_playlist_without_tracks_model,
+    playlist_model,
+)
+from .tracks import track, track_full
+
+
+class PurchasableContentItem(fields.Raw):
+    def format(self, value):
+        try:
+            if value.get("track_id"):
+                return marshal(value, track)
+            if value.get("playlist_id"):
+                return marshal(value, playlist_model)
+        except Exception as e:
+            raise MarshallingError(
+                "Unable to marshal as purchasable_content item"
+            ) from e
+
+
+class PurchasableContentItemFull(fields.Raw):
+    def format(self, value):
+        try:
+            if value.get("track_id"):
+                return marshal(value, track_full)
+            if value.get("playlist_id"):
+                return marshal(value, full_playlist_model)
+        except Exception as e:
+            raise MarshallingError(
+                "Unable to marshal as purchasable_content item"
+            ) from e
+
+
+purchasable_content_model = ns.model(
+    "purchasable_content",
+    {
+        "content_type": fields.String(enum=["track", "album"], required=True),
+        "item": PurchasableContentItem(required=True),
+        "class": fields.String(required=True, discriminator=True),
+    },
+)
+
+track_purchasable_content_model = ns.inherit(
+    "track_purchasable_content",
+    purchasable_content_model,
+    {
+        "content_type": fields.String(enum=["track"], required=True),
+        "item": fields.Nested(track, required=True),
+    },
+)
+
+collection_purchasable_content_model = ns.inherit(
+    "collection_purchasable_content",
+    purchasable_content_model,
+    {
+        "content_type": fields.String(enum=["album"], required=True),
+        "item": fields.Nested(playlist_model, required=True),
+    },
+)
+
+purchasable_content_full_model = ns.model(
+    "purchasable_content_full",
+    {
+        "content_type": fields.String(enum=["track", "album"], required=True),
+        "item": PurchasableContentItemFull(required=True),
+        "class": fields.String(required=True, discriminator=True),
+    },
+)
+
+track_purchasable_content_full_model = ns.inherit(
+    "track_purchasable_content_full",
+    purchasable_content_full_model,
+    {
+        "content_type": fields.String(enum=["track"], required=True),
+        "item": fields.Nested(track_full, required=True),
+    },
+)
+
+collection_purchasable_content_full_model = ns.inherit(
+    "collection_purchasable_content_full",
+    purchasable_content_full_model,
+    {
+        "content_type": fields.String(enum=["album"], required=True),
+        "item": fields.Nested(full_playlist_model, required=True),
```

Comment:
Based on what I saw in the query code, this is probably not the right model. This one will generate code that expects each playlist to contain a fully specified list of tracks (and will throw attempting to parse it).
If the query _does_ return nested tracks for each playlist, ignore me. Otherwise, swap this for `full_playlist_without_tracks_model`